service/matching/matcher.go

```diff
@@ -105,7 +103,20 @@ func (tm *TaskMatcher) Offer(ctx context.Context, task *internalTask) (bool, err
 			return true, err
 		}
 		return false, nil
-	default: // no poller waiting for tasks
+	default:
+		// no poller waiting for tasks, try forwarding this task to the
+		// root partition if available
+		select {
+		case token := <-tm.fwdrAddReqTokenC():
```

Comment:
first version of my code had the maxOutstanding logic within the forwarder. But it doesn't work because the channel needs to be exposed so that the caller can block on the channel along with other things (i.e. wait for a local poller or a forwarding token to be available simultaneously in a select{} block). The job of matcher is to still match producer with consumer. It now also enforces maxOutstanding / ratelimits